src/libsystemd-network/sd-radv.c

```diff
@@ -631,19 +632,21 @@ int sd_radv_add_prefix(sd_radv *ra, sd_radv_prefix *p) {
         LIST_APPEND(prefix, ra->prefixes, p);
         ra->n_prefixes++;
 
-        if (ra->state == RADV_STATE_IDLE) {
-                log_radv(ra, "Added prefix %s", strna(addr_p));
+        log_radv(ra, "Added prefix %s", strna(addr_p));
+
+        if (ra->state == RADV_STATE_IDLE)
+                return 0;
+
+announce:
```

Comment:
Maybe a small static helper instead of these gotos would look better?

Reply:
OK. I will do that in another PR.

Reply:
#21820